src/remote/util.py

```diff
@@ -133,7 +133,7 @@ def generate_command_str(self) -> str:
         """Generate the base ssh command to execute (without host)"""
         return prepare_shell_command(self.generate_command())
 
-    def execute(self, command: str, raise_on_error: bool = True) -> int:
+    def execute(self, command: str, raise_on_error: bool = True, control_master_none: bool = False) -> int:
```

Comment:
Instead of adding the use-case specific `control_master_none: bool = False` here, it would be better if you add a generic `extra_arguments: Optional[List[str] = None`. This way the API users will be able to add whatever they want here